pyqtgraph/colormap.py

```diff
@@ -255,24 +332,35 @@ def __init__(self, pos, color, mapping=CLIP, mode=None): #, names=None):
             The default of `ColorMap.CLIP` continues to show
             the colors assigned to 0 and 1 for all values below or above this range, respectively.
         """
+        self.name = name # storing a name helps identify ColorMaps sampled by Palette
         if mode is not None:
             warnings.warn(
                 "'mode' argument is deprecated and does nothing.",
                 DeprecationWarning, stacklevel=2
         )
-        self.pos = np.array(pos)
-        order = np.argsort(self.pos)
-        self.pos = self.pos[order]
-        self.color = np.apply_along_axis(
-            func1d = lambda x: np.uint8( mkColor(x).getRgb() ), # cast RGB integer values to uint8
-            axis   = -1,
-            arr    = color,
-            )[order]
+        if pos is None:
+            order = range(len(color))
+            self.pos = np.linspace(0.0, 1.0, num=len(color))
+        else:
+            self.pos = np.array(pos)
+            order = np.argsort(self.pos)
+            self.pos = self.pos[order]
+        
+        self.color = np.zeros( (len(color), 4) ) # stores float rgba values
+        for cnt, idx in enumerate(order):
+            self.color[cnt] = mkColor(color[idx]).getRgbF()
+        # alternative code may be more efficinet, but fails to handle lists of QColor.
```

Comment:
totally good with this kind of code being left in-place but commented out 👍🏻

Reply:
The previous code was quite elegant, but it silently assumed a 2-dimensional list of values.
It seems like it should be possible to adapt it to handle a 1-dimensional list of ready-made colors, too, but I just couldn't figure it out. Maybe someone else wants to have a go; so I would propose to leave this in for reference.